### PR TITLE
Add resources for Samsung Galaxy A54 5G (a54x)

### PR DESCRIPTION
**Pull request by Pokey8680**
# Changes
- Add power profile to prevent unneccesary power usage/battery drain
- Status bar height fixed
- Android respects the presence of the hole punch
- Status bar elements bottom aligned, similar to stock firmware
- Rounded corner padding fixed
- Rounded corner padding more accurate
- Fix Always-On Display (AOD)
  - Wouldn't refresh before, now it does
  - Enabled by default
- Set default refresh rate to 120hz
- Enable burn-in protection by shifting elements of the UI (Like on stock firmware)
- Enable 25W fast charging
- Enable dynamic refresh rate
  - Lowers refresh rate from 120hz to 60hz when the content is static, to save battery

### CODE REVIEW

**Comment by Pokey8680** [comment]
Sorry for deleting and remaking pull requests. this is supposed to be my last one until it gets merged into the main Trebledroid repo. Forked a new repo to clean up useless commits.

File: Samsung/a54x-SystemUI/res/values/config.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="status_bar_padding_start">40px</dimen>
+    <dimen name="status_bar_padding_top">35px</dimen>
+    <dimen name="physical_power_button_center_screen_location_y">1150px</dimen>
+    <dimen name="rounded_corner_radius_bottom">50.0dip</dimen>
+    <dimen name="rounded_corner_radius_top">50.0dip</dimen>
+    <dimen name="status_bar_height_portrait">6.779999mm</dimen>
+    <string name="config_mainBuiltInDisplayCutout">M 0,0 M 0,7.822222222222222‬‬ a 10.31111111111111‬,10.31111111111111‬ 0 1,0 0,20.62222222222222‬ a 10.31111111111111‬,10.31111111111111‬ 0 1,0 0,-20.62222222222222‬ Z @dp</string>
```
Comment (ponces) [inline Samsung/a54x-SystemUI/res/values/config.xml]: Remove `config_mainBuiltInDisplayCutout` from the SystemUI overlay

File: Samsung/a54x/res/values/config.xml [hunk 2]
```diff
@@ -0,0 +1,2260 @@
...
+    <dimen name="cover_clear_view_bottom">242.0dip</dimen>
+    <dimen name="cover_clear_view_left">204.0dip</dimen>
+    <dimen name="cover_clear_view_right">360.0dip</dimen>
+    <dimen name="cover_clear_view_rounded_corner">33.0dip</dimen>
+    <dimen name="cover_clear_view_top">1.0dip</dimen>
+    <dimen name="status_bar_camera_side_padding">30.0px</dimen>
+    <dimen name="status_bar_camera_top_margin">22.0px</dimen>
+    <dimen name="status_bar_height_portrait">80.0px</dimen>
+    <dimen name="status_bar_indicator_corner_padding">60.0px</dimen>
+      <string name="config_mainBuiltInDisplayCutout">M 0,0 M 0,7.822222222222222‬‬ a 10.31111111111111‬,10.31111111111111‬ 0 1,0 0,20.62222222222222‬ a 10.31111111111111‬,10.31111111111111‬ 0 1,0 0,-20.62222222222222‬ Z @dp</string>
```
Comment (ponces) [inline Samsung/a54x/res/values/config.xml]: Remove hidden unicode chars

File: Samsung/a54x/res/values/config.xml [hunk 3]
```diff
@@ -0,0 +1,2260 @@
...
+      
+       <integer name="config_longPressOnPowerBehavior">1</integer>
+    <integer name="config_multiuserMaximumUsers">4</integer>
+    <integer name="config_defaultRefreshRate">0</integer>
+      <integer name="config_defaultPeakRefreshRate">120</integer>
+    
+    <string name="config_mobile_hotspot_provision_app_no_ui">com.samsung.provisioning.SILENT_PROVISIONING_CHECK</string>
```
Comment (ponces) [inline Samsung/a54x/res/values/config.xml:2257]: Remove this

File: Samsung/a54x/res/values/config.xml [hunk 4]
```diff
@@ -0,0 +1,2260 @@
...
+    <string name="config_mobile_hotspot_provision_response">com.samsung.provisioning.SILENT_PROVISIONING_CHECK_RESULT</string>
```
Comment (ponces) [inline Samsung/a54x/res/values/config.xml:2258]: Remove this

File: overlay.mk [hunk 5]
```diff
@@ -251,6 +251,8 @@ PRODUCT_PACKAGES += \
 	treble-overlay-samsung-a52 \
 	treble-overlay-samsung-a53 \
 	treble-overlay-samsung-a53-systemui \
+ treble-overlay-samsung-a54x \
```
Comment (ponces) [inline overlay.mk]: Add indentation

File: overlay.mk [hunk 6]
```diff
@@ -251,6 +251,8 @@ PRODUCT_PACKAGES += \
...
+ treble-overlay-samsung-a54x-systemui \
```
Comment (ponces) [inline overlay.mk]: Add indentation

**Review by ponces — CHANGES_REQUESTED** [review]
Run the test script and remove all unneeded configs

File: Samsung/a54x/res/values/config.xml (hunk 2 above)
Comment (Pokey8680) [reply Samsung/a54x/res/values/config.xml]: I can't seem to remove them no matter what I try.

Comment (Pokey8680) [reply Samsung/a54x/res/values/config.xml]: UPDATE: I managed to remove the hidden Unicode here.

**Comment by Pokey8680** [comment]
Done

**Review by ponces — CHANGES_REQUESTED** [review]
Have you really run the test script?

```
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_High_Brightness_Mode_Plus_Value
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_HysteresisBrightInputLux
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_HysteresisBrightOutputLux
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_HysteresisDarkInputLux
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_HysteresisDarkOutputLux
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_Screen_Brightness_Backlight_Value
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_Screen_Brightness_Nits_Value
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_aodBrightnessValues
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_mobile_hotspot_provision_app
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_screenBrightnessRangeForClearView
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_dynamic_automatic_brightness_available
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_systemCaptionsServiceCallsEnabled
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_enableExynosDisplaySolutionService
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_enableWifiDisplay
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_flipToScreenOffEnabled
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_screenBrightnessDimFloat
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute cover_clear_view_bottom
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute cover_clear_view_left
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute cover_clear_view_right
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute cover_clear_view_rounded_corner
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute cover_clear_view_top
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute status_bar_camera_side_padding
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute status_bar_camera_top_margin
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute status_bar_indicator_corner_padding
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_longPressOnPowerBehavior
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_mobile_hotspot_provision_app_no_ui
Fatal: ./Samsung/a54x/res/values/config.xml: defines a non-existing attribute config_mobile_hotspot_provision_response
```

**Comment by Pokey8680** [comment]
It didn't show those when I ran the script, but I've removed those values now.

**Review by ponces — APPROVED** [review]

**Pull request merged**